### PR TITLE
Query: Entity splitting support for regular entities

### PR DESCRIPTION
**Pull request by smitpatel**
Part of #620

### CODE REVIEW

File: src/EFCore.Relational/Query/SqlExpressions/SelectExpression.cs [hunk 1]
```diff
@@ -314,47 +314,106 @@ internal SelectExpression(IEntityType entityType, ISqlExpressionFactory sqlExpre
...
+                    {
+                        var table = mappings[0].Table;
 
-                var tableReferenceExpression = new TableReferenceExpression(this, tableExpression.Alias!);
-                AddTable(tableExpression, tableReferenceExpression);
+                        GenerateNonHierarchyNonSplittingEntityType(table, new TableExpression(table));
+                    }
+                    else
+                    {
+                        // table splitting
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Query/SqlExpressions/SelectExpression.cs:335]:
```suggestion
                        // entity splitting
```

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs [hunk 2]
```diff
@@ -596,7 +596,7 @@ public override void QF_Select_Correlated_Subquery_In_Anonymous_Nested_With_QF()
             @"SELECT [o].[CustomerId], [o].[OrderDate]
 FROM [Orders] AS [o]
 INNER JOIN (
-    SELECT [g].[OrderId]
+    SELECT [c].[Id], [c].[FirstName], [c].[LastName], [g].[OrderId], [g].[CustomerId], [g].[OrderDate]
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: Why are we selecting additional column now?

Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: side-effect of removing join tables
Earlier we did base processing for inner join which identified the column being used in join predicate and removed others if not used. The code for removal of join table, excludes the join predicate columns too so we can be sure to remove the table but as a side-effect, now we don't identify the columns used if none used.
This only affects the case where there is a inner join which is not contributing anything to result. If there is any single column used in actual result from inner join, we will trim it down.

Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: Same effect existed for Left join from TPT, we can consider in future if this is significant scenario for us to do something about it.

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: Are you planning to address this case in the future?

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: File an issue

File: test/EFCore.SqlServer.FunctionalTests/EntitySplittingSqlServerTest.cs [hunk 3]
```diff
@@ -62,6 +62,33 @@ await using (var context = CreateContext())
         }
     }
 
+    public override async Task Can_roundtrip()
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/EntitySplittingSqlServerTest.cs:65]: Also unskip `Can_roundtrip_with_triggers`

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs (hunk 2 above)
Comment (smitpatel) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs:599]: #28426

**Pull request merged**